pkg/cloudprovider/providers/rackspace/rackspace.go

```diff
@@ -549,23 +549,21 @@ func (os *Rackspace) Routes() (cloudprovider.Routes, bool) {
 }
 
 func (os *Rackspace) GetZone() (cloudprovider.Zone, error) {
-	glog.V(1).Infof("Current zone is %v", os.region)
-
 	return cloudprovider.Zone{Region: os.region}, nil
 }
 
 // GetZoneByProviderID implements Zones.GetZoneByProviderID
 // This is particularly useful in external cloud providers where the kubelet
 // does not initialize node data.
 func (os *Rackspace) GetZoneByProviderID(providerID string) (cloudprovider.Zone, error) {
-	return cloudprovider.Zone{}, errors.New("GetZoneByProviderID not implemented")
+	return cloudprovider.Zone{Region: os.region}, nil
```

Comment:
Please utilize providerID in these methods.

If providerID does not affect the region variable, then please add a comment detailing why

Reply:
Done.

Reply:
I am sorry for my mistake, we should get zone from server by providerID or NodeName.